numismatic/feeds/base.py

```diff
@@ -71,8 +73,18 @@ class WebsocketClient(abc.ABC):
     raw_stream = attr.ib(default=None)
     raw_interval = attr.ib(default=1)
 
+    @classmethod
+    async def _connect(cls, wss_url=None):
+        if wss_url is None:
+            wss_url = cls.wss_url
+        logger.info(f'Connecting to {wss_url!r} ...')
+        ws = await websockets.connect(wss_url)
+        if hasattr(cls, 'on_connect'):
```

Comment:
Shouldn't this be an abstract method on the class, and if so, we wouldn't need to check using `hasattr`? The only reason it would not need to be abstract method is if it were sometimes not used, but this seems like something that would be used for every connection???